trio/_file_io.py

```diff
@@ -125,18 +278,169 @@ async def aclose(self):
+        def errors(self: AsyncIOWrapper[_HasErrors]) -> str | None: ...
+        @property
+        def newlines(self: AsyncIOWrapper[_HasNewlines]) -> Any: ...  # None, str or tuple
+        @property
+        def buffer(self: AsyncIOWrapper[_HasBuffer]) -> BinaryIO: ...
+        @property
+        def raw(self: AsyncIOWrapper[_HasRaw]) -> io.RawIOBase: ...
+        @property
+        def line_buffering(self: AsyncIOWrapper[_HasLineBuffering]) -> int: ...
+        @property
+        def closefd(self: AsyncIOWrapper[_HasCloseFD]) -> bool: ...
+        @property
+        def name(self: AsyncIOWrapper[_HasName]) -> str: ...
+        @property
+        def mode(self: AsyncIOWrapper[_HasMode]) -> str: ...
+
+        def fileno(self: AsyncIOWrapper[_HasFileNo]) -> int: ...
+        def isatty(self: AsyncIOWrapper[_HasIsATTY]) -> bool: ...
+        def readable(self: AsyncIOWrapper[_HasReadable]) -> bool: ...
+        def seekable(self: AsyncIOWrapper[_HasSeekable]) -> bool: ...
+        def writable(self: AsyncIOWrapper[_HasWritable]) -> bool: ...
+        def getvalue(self: AsyncIOWrapper[_CanGetValue[AnyStr]]) -> AnyStr: ...
+        def getbuffer(self: AsyncIOWrapper[_CanGetBuffer]) -> memoryview: ...
+        async def flush(self: AsyncIOWrapper[_CanFlush]) -> None: ...
+        async def read(self: AsyncIOWrapper[_CanRead[AnyStr]], __size: int | None = -1) -> AnyStr: ...
+        async def read1(self: AsyncIOWrapper[_CanRead1], __size: int | None = -1) -> bytes: ...
+        async def readall(self: AsyncIOWrapper[_CanReadAll[AnyStr]]) -> AnyStr: ...
+        async def readinto(self: AsyncIOWrapper[_CanReadInto], __buf: Buffer) -> int | None: ...
+        async def readline(self: AsyncIOWrapper[_CanReadLine[AnyStr]], __size: int = -1) -> AnyStr: ...
+        async def readlines(self: AsyncIOWrapper[_CanReadLines[AnyStr]]) -> list[AnyStr]: ...
+        async def seek(self: AsyncIOWrapper[_CanSeek], __target: int, __whence: int = 0) -> int: ...
+        async def tell(self: AsyncIOWrapper[_CanTell]) -> int: ...
+        async def truncate(self: AsyncIOWrapper[_CanTruncate], __size: int | None = None) -> int: ...
+        async def write(self: AsyncIOWrapper[_CanWrite[AnyStr]], __data: AnyStr) -> int: ...
+        async def writelines(self: AsyncIOWrapper[_CanWriteLines[T]], __lines: Iterable[T]) -> None: ...
+        async def readinto1(self: AsyncIOWrapper[_CanReadInto1], __buffer: Buffer) -> int: ...
+        async def peek(self: AsyncIOWrapper[_CanPeek[AnyStr]], __size: int = 0) -> AnyStr: ...
+
+
+# Type hints are copied from builtin open.
+_OpenFile = Union["StrOrBytesPath", int]
+_Opener = Callable[[str, int], int]
+
+
+@overload
+async def open_file(
+    file: _OpenFile,
+    mode: OpenTextMode = "r",
+    buffering: int = -1,
+    encoding: str | None = None,
+    errors: str | None = None,
+    newline: str | None = None,
+    closefd: bool = True,
+    opener: _Opener | None = None,
+) -> AsyncIOWrapper[io.TextIOWrapper]:
+    ...
```

Comment:
Let's get a separate single-line PR that quickly adds `@overload` to `exclude_lines` in `.coveragerc`.

Reply:
This was done in this PR.